src/Action/Task/GenerateOrders.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace AppBundle\Action\Task;
+
+use AppBundle\Entity\Sylius\Order;
+use AppBundle\Entity\Task;
+use AppBundle\Pricing\PricingManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Recurr\Transformer\ArrayTransformer;
+use Recurr\Transformer\Constraint\BetweenConstraint;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+class GenerateOrders
+{
+    public function __construct(
+        private readonly EntityManagerInterface $entityManager,
+        private readonly PricingManager $pricingManager)
+    {
+    }
+
+    public function __invoke($data, Request $request)
+    {
+        //get query parameters
+        $queryParams = $request->query->all();
+
+        $date = $queryParams['date'];
+
+        if (empty($date)) {
+            throw new BadRequestHttpException('Date is required');
+        }
+
+        $allSubscriptions = $this->entityManager->getRepository(Task\RecurrenceRule::class)->findBy(['deletedAt' => null]);
+
+        $subscriptions = array_filter($allSubscriptions, function ($subscription) use ($date) {
+            return $this->filterByDate($subscription, $date);
+        });
+
+        $subscriptions = array_filter($subscriptions, function ($subscription) use ($date) {
+            return $this->filterWithoutOrdersOnDate($subscription, $date);
+        });
+
+        $orders = [];
+
+        foreach ($subscriptions as $subscription) {
+            $order = $this->pricingManager->createOrderFromSubscription($subscription, $date);
+            if (null !== $order) {
+                $orders[] = $order;
+            }
+        }
+
+        return $orders;
+    }
+
+    private function filterByDate(Task\RecurrenceRule $subscription, string $startDate): bool
+    {
+        $after = new \DateTime($startDate . ' 00:00');
+        $before = new \DateTime($startDate . ' 23:59');
+
+        $transformer = new ArrayTransformer();
+        $constraint = new BetweenConstraint(
+            $after,
+            $before,
+            $inc = true
+        );
+
+        $rule = $subscription->getRule();
+
+        $rule->setStartDate($after);
+        $rule->setEndDate($before);
+
+        $occurrences = $transformer->transform($rule, $constraint);
+
+        return count($occurrences) > 0;
+    }
+
+    private function filterWithoutOrdersOnDate(Task\RecurrenceRule $subscription, string $startDate): bool
+    {
+        $date = new \DateTime($startDate . ' 00:00');
+
+        $orders = $this->entityManager->getRepository(Order::class)->findBySubscriptionAndDate($subscription, $date);
+
+        // Ideally, we should create an order for each subscription,
```

Comment:
hello if I understand well you are creating also the tasks linked to recurrence rules as well here? i guess in this case you should also remove the button that enable generating manually recurring task in the RecurrenceRuleModalContent (the old one)

Reply:
For subscriptions: I create an order + a delivery + tasks.

For the existing recurring rules I'll do the same, but only if there is more than one task and the first one is a pickup (this is the existing behaviour: https://github.com/coopcycle/coopcycle-web/blob/master/src/Action/Task/RecurrenceRuleBetween.php#L85). If it's not the case only tasks are created.

So here it's just a check to avoid an attempt to create an order if there are already tasks linked to a subscription.

The idea is to remove current RecurrenceRuleModalContent along with `Generate tasks` button, but we'll discuss it on GA first